src/test/java/io/cdap/plugin/gcp/bigquery/sink/AvroOutputFormatTest.java

```diff
@@ -0,0 +1,55 @@
+package io.cdap.plugin.gcp.bigquery.sink;
+
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.hadoop.io.AvroSerialization;
+import org.apache.avro.mapred.AvroKey;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.hamcrest.beans.SamePropertyValuesAs;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.function.Supplier;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(AvroSerialization.class)
+public class AvroOutputFormatTest {
+
+  @Mock
+  TaskAttemptContext context;
+
+  @Mock
+  Configuration mockConf;
+
+  @Mock
+  GenericData mockDataModel;
+
+  @Test
+  public void testRecordWriterCreation() throws IOException {
+    Mockito.when(context.getConfiguration()).thenReturn(mockConf);
+    PowerMockito.mockStatic(AvroSerialization.class);
+    PowerMockito.when(AvroSerialization.createDataModel(mockConf)).thenReturn(mockDataModel);
+
+    AvroOutputFormat avroOutputFormat = new AvroOutputFormat();
+
+    RecordWriter<AvroKey<GenericRecord>, NullWritable> actual =
+      avroOutputFormat.getRecordWriter(context);
+    AvroRecordWriter avroRecordWriter = (AvroRecordWriter) actual;
+
+    Supplier<OutputStream> outputStreamSupplier = avroOutputFormat.getOutputStreamSupplier(context);
+
+    Assert.assertThat(outputStreamSupplier,
+                      SamePropertyValuesAs.samePropertyValuesAs(avroRecordWriter.getOutputStreamSupplier()));
```

Comment:
I don't think this does what you want. It would make sense to do `samePropertyValuesAs` for the `avroRecordWriter` as it has the property. The `Supplier` has no proeprties so that's essentially a NOP check. Also I am not sure what are you trying to test here? Why do you care that those are same or not?
Probably you care that it writes to a proper file. But in this case, just get the output stream and validate where does it write to.

Reply:
Just wanted to have a unit test for the code change, it doesn't do anything significant. 
I will remove this unit test.